samples/soundtest.lobster

```diff
@@ -0,0 +1,173 @@
+            gl_scale(scale)
+
+            let dir = xy_f { (gl_button("d") >= 1) - (gl_button("a") >= 1),
+                          (gl_button("s") >= 1) - (gl_button("w") >= 1) }
+            let newpos = playerpos + normalize(dir) * delta_time_w_pause() * playerspeed
+            if !any(abs(newpos) > float(gl_window_size()) / scale / 2):
+                playerpos = newpos
+
+            let tomouse = normalize(gl_local_mouse_pos(0) - playerpos)
+
+            if !paused:
+                if lastenemy < t_gl:
+                    enemies.push(enemy { sincos(rnd(360)) * worldsize * 2, enemymaxhp })
+                    lastenemy += enemyrate
+                    enemyrate *= 0.999
+
+                if lastbullet < t_gl:
+                    bullets.push(bullet { playerpos, tomouse })
+                    lastbullet += firerate
+
+            closest_enemy_dist = 9999.0
+            for(enemies) e:
+                let playervec = playerpos - e.pos
+                let playerdist = magnitude(playervec)
+                let playerdir = normalize(playervec)
+                if playerdist < closest_enemy_dist:
+                    closest_enemy_dist = playerdist
+                e.pos += playerdir * delta_time_w_pause() * enemyspeed
+                for(bullets) b:
+                    if magnitude(b.pos - e.pos) < 1:
+                        e.hp--
+                        if(e.hp <= 0):
+                            e.hp = 0
+                            score++
+                        b.pos = xy_x * worldsize * 10
+                if playerdist < 1:
+                    playerhealth -= delta_time_w_pause() * 50
+                    if playerhealth <= 0:
+                        playerhealth = 0
+                        highscore = max(highscore, score)
+                        playing = false
+                gl_color lerp(color_red, color_blue, div(e.hp, enemymaxhp)):
+                    renderpointytriangle(e.pos, playerdir)
+            if !paused:
+                // distance-dependent UFO buzz sound
+                let enemy_volume = clamp(1.0/(closest_enemy_dist*closest_enemy_dist + 0.001), 0.0, 1.0)
+                sound_volume(ufobuzz, enemy_volume)
+
+            enemies = filter(enemies): _.hp
+
+            for(bullets) b:
+                b.pos += b.dir * delta_time_w_pause() * bulletspeed
+                gl_translate b.pos:
+                    gl_color color_yellow:
+                        gl_circle(0.2, 20)
+
+            bullets = filter(bullets) b:
+                magnitude(b.pos) < worldsize * 2
+
+            gl_translate gl_local_mouse_pos(0):
+                gl_line_mode 1:
+                    gl_color color_grey:
+                        gl_circle(0.5, 20)
+
+            renderpointytriangle(playerpos, tomouse)
+    else:
+        let msg = "press space to play!"
+        gl_translate float(gl_window_size() - gl_text_size(msg)) / 2:
+            gl_text(msg)
+        if not paused:
```

Comment:
this is additional logic that shouldn't need to be here.

Reply:
Mmmh. If you die, I want the music to stop again like it was in the beginning. This is the purpose of this code block. However, I should have init'ed paused to true.